docs/en/guides/pipeline-integrations/development/sgtk-developer-bootstrapping.md

```diff
@@ -0,0 +1,409 @@
+# case, programmatically.
+from tank_vendor.shotgun_authentication import ShotgunAuthenticator
+
+# Instantiate the CoreDefaultsManager. This allows the ShotgunAuthenticator to
+# retrieve the site, proxy and optional script_user credentials from shotgun.yml
+cdm = sgtk.util.CoreDefaultsManager()
+
+# Instantiate the authenticator object, passing in the defaults manager.
+authenticator = ShotgunAuthenticator(cdm)
+
+# Create a user programmatically using the script's key.
+user = authenticator.create_script_user(
+ api_script="Script Name",
+ api_key="4e48f....<use the key from your Shotgun site>"
+)
+
+# Tells Toolkit which user to use for connecting to Shotgun.
+sgtk.set_authenticated_user(user)
+
+# Bootstrap
+###########
+
+# create an instance of the ToolkitManager which we will use to set a bunch of settings before initiating the bootstrap. 
+mgr = sgtk.bootstrap.ToolkitManager()
+mgr.plugin_id = "basic.shell"
+
+project = {"type": "Project", "id": 176}
+
+engine = mgr.bootstrap_engine("tk-shell", entity=project)
+```
+
+## Part 5 - Launching an app
+
+Now you have an engine instance, you're ready to start using the Toolkit API.
+
+Before covering how to launch the app, it's worth pointing out you can get hold of the [current context](https://developer.shotgunsoftware.com/tk-core/platform.html#sgtk.platform.Engine.context), [Sgtk instance](https://developer.shotgunsoftware.com/tk-core/platform.html#sgtk.platform.Engine.sgtk), and [Shotgun API instance](https://developer.shotgunsoftware.com/tk-core/platform.html#sgtk.platform.Engine.shotgun) via the engine.
+
+```python
+engine.context
+engine.sgtk
+engine.shotgun
+```
+With these, you could instead of launching an app, test code snippets, or run some automation requiring the Toolkit API.
+
+### Launching the App
+
+When the engine starts, it initializes all the apps defined for the environment. 
+The apps in turn register commands with the engine, and the engine usually displays these as actions in a menu, 
+if running a Software like Maya.
+
+#### Finding the commands
+To first see what commands have been registered you can print our the [`engine.commands`](https://developer.shotgunsoftware.com/tk-core/platform.html#sgtk.platform.Engine.commands)
+
+```python
+# use pprint to give us a nicely formatted output.
+import pprint
+pprint.pprint(engine.commands.keys())
+
+>> ['houdini_fx_17.5.360',
+ 'nukestudio_11.2v5',
+ 'nukestudio_11.3v2',
+ 'after_effects_cc_2019',
+ 'maya_2019',
+ 'maya_2018',
+ 'Jump to Screening Room Web Player',
+ 'Publish...',
+...]
+```
+
+With that list, you can see which commands have been registered and can be run.
+
+#### Running the command
+
+How you run the command will be different depending on the engine, as there is currently no standardized method.
+For the `tk-shell` engine you can use `engine.execute_command` which expects a command string name, which we listed out earlier,
+ and a list of parameters that the app's command expects to be passed.
```

Comment:
The thing I'm not keen on doing with adding `()` to the end of method, is that suggests a call and if you are calling it you should provide the arguments. To me writing `engine.execute_command()` suggests that it can be called with no arguments which is not true. I'm not sure if that is standard practice/convention to do it anyway though.